File: SH/etc/121. Best_Time_to_Buy_and_Sell_Stock.c
int maxProfit(int* prices, int pricesSize) {
    int profit = 0;
    int buy = prices[0];
    for(int i=1; i<pricesSize; i++){
        int sell = prices[i];
        int potential = sell - buy;
        if(buy>sell){buy = sell;}
        if(potential>profit){profit=potential;}
    }
    return profit;
}

File: SH/etc/121. Best_Time_to_Buy_and_Sell_Stock.py
def maxProfit(prices:List[int])-> int:
    profit = 0
    buy = prices[0]
    for i in range(1, len(prices)):
        sell = prices[i]
        potential = sell - buy
        if buy > sell:
            buy = sell
        elif potential > profit:
            profit = potential
    return profit